### PR TITLE
Added document line numbers to CoNLL parsed dataframes

### PR DESCRIPTION
**Pull request by BryanCutler**
Fixes https://github.ibm.com/frreiss/conll_relabel_paper/issues/27

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @xuhdev @ZachEichen

File: text_extensions_for_pandas/io/conll.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def add_line(self, line_num: int, line_elems: List[str]):
 def _parse_conll_file(input_file: str,
                       column_names: List[str],
                       iob_columns: List[bool]) \
-        -> List[List[_SentenceData]]:
+        -> (List[List[_SentenceData]], List[List[int]]):
```
Comment (frreiss) [inline text_extensions_for_pandas/io/conll.py:141]: Can you please fold these line numbers into the existing `_SentenceData` data structure? Then the return type of this function won't need to be a tuple.

File: text_extensions_for_pandas/io/conll.py [hunk 2]
```diff
@@ -343,7 +352,8 @@ def _iob_to_iob2(df: pd.DataFrame, column_names: List[str],
 def _doc_to_df(doc: List[_SentenceData],
                column_names: List[str],
                iob_columns: List[bool],
-               space_before_punct: bool) -> pd.DataFrame:
+               space_before_punct: bool,
+               doc_line_nums: List[int]) -> pd.DataFrame:
```
Comment (frreiss) [inline text_extensions_for_pandas/io/conll.py:356]: This additional argument can go away if line numbers are included in the `_SentenceData` data structure.

File: text_extensions_for_pandas/io/conll.py [hunk 3]
```diff
@@ -357,11 +367,11 @@ def _doc_to_df(doc: List[_SentenceData],
      the returned dataframe will contain *two* columns, holding IOB2 tags and
      entity type tags, respectively. For example, an input column "ent" will turn into
      output columns "ent_iob" and "ent_type".
-
     :param space_before_punct: If `True`, add whitespace before
      punctuation characters (and after left parentheses)
      when reconstructing the text of the document.
-    :return: DataFrame with four columns:
+    :param doc_line_nums: list of line numbers in parsed file for each token in the doc
```
Comment (frreiss) [inline text_extensions_for_pandas/io/conll.py:373]: This additional argument can go away if line numbers are included in the `_SentenceData` data structure.

File: text_extensions_for_pandas/io/conll.py [hunk 4]
```diff
@@ -700,9 +712,9 @@ def conll_2003_to_dataframes(input_file: str,
     * `ent_type`: Entity type names for tokens tagged "I" or "B" in
       the `ent_iob` column; `None` everywhere else.
     """
-    parsed_docs = _parse_conll_file(input_file, column_names, iob_columns)
-    doc_dfs = [_doc_to_df(d, column_names, iob_columns, space_before_punct)
-               for d in parsed_docs]
+    parsed_docs, doc_line_nums = _parse_conll_file(input_file, column_names, iob_columns)
```
Comment (frreiss) [inline text_extensions_for_pandas/io/conll.py:715]: This changes on these 3 lines can go away if line numbers are included in the `_SentenceData` data structure.

**Comment by BryanCutler** [comment]
Thanks @frreiss , I was going to add them to SentenceData, but thought it didn't quite seem to belong. I agree it would simplify this change quite a bit though, so I'll give it a shot.

**Comment by frreiss** [comment]
LGTM. Thanks @BryanCutler !

**Pull request merged**